src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -482,7 +517,12 @@ public void setNetworkTimeout(Executor executor, int milliseconds) throws SQLExc
     @Override
     public int getNetworkTimeout() throws SQLException {
         checkClosed();
-        throw new ReadReplicaUnsupportedOperationException();
+        if(compatibleWithPreviousVersion)
+            throw new ReadReplicaUnsupportedOperationException();
+        else
+            return connectionProvider
+                .getReadConnection(new RouteDecisionBuilder(Reason.RO_API_CALL))
```

Comment:
You didn't add tests, So I guess that you choose to add `RO_API_CALL` `RW_API_CALL` based on some product usage. Can you add tests ?

Reply:
What would you like to test here? 🤔

Reply:
We can test if the delegate has been called, as the implementation seems to be just a delegate. Otherwise, the method will not be test-covered. Not sure if it's worth it.

Reply:
We also didn't integration-test every single JDBC method, because we don't have a JDBC/SQL workload covering all these APIs.

Reply:
Tests based on implementation are bad practice or at best very low value. It makes implementation harder to change, even when logic stays the same.